src/Resources/Resources/RoleAssignments/SetAzureRoleAssignmentCommand.cs

```diff
@@ -0,0 +1,130 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ActiveDirectory;
+using Microsoft.Azure.Commands.Resources.Models;
+using Microsoft.Azure.Commands.Resources.Models.Authorization;
+using Microsoft.Azure.Management.Internal.Network.Version2017_10_01.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Resources
+{
+    /// <summary>
+    /// Updates an existing role assignment.
+    /// </summary>
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RoleAssignment", DefaultParameterSetName = ParameterSet.RoleAssignment), OutputType(typeof(PSRoleAssignment))]
```

Comment:
Sorry I forgot to mention in the design review that all the cmdlets that make change on azure should support "ShouldProcess", which basically means you should declare `SupportShouldProcess` in the attribute and call `ConfirmAction()` in `ExecuteCmdlet()`. You can take `Remove-AzRoleAssignment` as an example: https://github.com/Azure/azure-powershell/blob/fee5a593ce9437f8796de39f25ecce1cad1115c1/src/Resources/Resources/RoleAssignments/RemoveAzureRoleAssignmentCommand.cs